auv_teleop/scripts/vehicle_control_tab.py

```diff
@@ -0,0 +1,216 @@
+        self.teleop_stop.clicked.connect(self.stop_teleop)
+        self.start_enable.clicked.connect(self.start_enable_publishing)
+        self.stop_enable.clicked.connect(self.stop_enable_publishing)
+
+    def connect_publishing_button(self, button, direction, speed_getter):
+        button.pressed.connect(lambda: self.start_publishing(direction, speed_getter()))
+        button.released.connect(self.stop_publishing)
+
+    def get_linear_spinbox(self):
+        return self.linear_speed_spinbox.value()
+
+    def get_angular_spinbox(self):
+        return self.angular_speed_spinbox.value()
+
+    def start_teleop(self):
+        cmd = ["roslaunch", "auv_teleop", "start_teleop.launch"]
+        if self.xbox_check.isChecked():
+            cmd.append("controller:=xbox")
+        print(f"Executing: {' '.join(cmd)}")
+        self.launch_process = subprocess.Popen(cmd)
+
+    def stop_teleop(self):
+        if self.launch_process is not None:
+            print("Terminating teleop process...")
+            self.launch_process.terminate()
+            try:
+                self.launch_process.wait(timeout=2)
+            except subprocess.TimeoutExpired:
+                print("Process did not terminate, killing it...")
+                self.launch_process.kill()
+            self.launch_process = None
+        else:
+            print("No teleop process to stop.")
+
+    def publish_velocity(self):
+        rate = rospy.Rate(20)
+        while self.publishing and not rospy.is_shutdown():
+            self.cmd_vel_pub.publish(self.current_twist)
+            rate.sleep()
+
+    def start_publishing(self, direction, speed):
+        self.publishing = True
+        self.current_twist = Twist()
+
+        direction_mapping = {
+            "forward": lambda: setattr(self.current_twist.linear, "x", speed),
+            "backward": lambda: setattr(self.current_twist.linear, "x", -speed),
+            "left": lambda: setattr(self.current_twist.linear, "y", speed),
+            "right": lambda: setattr(self.current_twist.linear, "y", -speed),
+            "pos_z": lambda: setattr(self.current_twist.linear, "z", speed),
+            "neg_z": lambda: setattr(self.current_twist.linear, "z", -speed),
+            "pos_yaw": lambda: setattr(self.current_twist.angular, "z", speed),
+            "neg_yaw": lambda: setattr(self.current_twist.angular, "z", -speed),
+        }
+
+        action = direction_mapping.get(direction)
+        if action:
+            action()
+        else:
+            print(f"Unknown direction: {direction}")
+
+        if not hasattr(self, "publish_thread") or not self.publish_thread.is_alive():
+            self.publish_thread = threading.Thread(target=self.publish_velocity)
+            self.publish_thread.start()
+
+    def stop_publishing(self):
+        self.publishing = False
+        self.current_twist = Twist()
+        self.cmd_vel_pub.publish(self.current_twist)
+
+    def publish_enable(self):
```

Comment:
Again lets add _loop to clarify its role as a running thread. And make sure everybody understands its about enabling control. And _signal will make it clear its a boolean on-off signal to enable/disable controllers
```suggestion
    def publish_control_enable_signal_loop(self):
```